ui/details-panel/FailureSummaryTab.jsx

```diff
@@ -0,0 +1,263 @@
+              onClick={() => fileBug(index)}
+              title="file a bug for this failure"
+            >
+              <i className="fa fa-bug" />
+            </span>}
+          <span>{suggestion.search}</span>
+        </div>
+
+        {/* <!--Open recent bugs--> */}
+        {suggestion.valid_open_recent &&
+        <ul className="list-unstyled failure-summary-bugs">
+          {suggestion.bugs.open_recent.map(bug =>
+            (<BugListItem
+              key={bug.id}
+              bug={bug}
+              selectedJob={selectedJob}
+              pinboardService={pinboardService}
+              suggestion={suggestion}
+              $timeout={$timeout}
+            />))}
+
+        </ul>}
+
+        {/* <!--All other bugs--> */}
+        {suggestion.valid_all_others && suggestion.valid_open_recent &&
+        <span
+          rel="noopener"
+          onClick={this.clickShowMore}
+          className="show-hide-more"
+        >Show / Hide more</span>}
+
+        {suggestion.valid_all_others && (suggestionShowMore
+          || !suggestion.valid_open_recent) &&
+          <ul className="list-unstyled failure-summary-bugs">
+            {suggestion.bugs.all_others.map(bug =>
+              (<BugListItem
+                key={bug.id}
+                bug={bug}
+                selectedJob={selectedJob}
+                pinboardService={pinboardService}
+                suggestion={suggestion}
+                $timeout={$timeout}
+                bugClassName={bug.resolution !== "" ? "deleted" : ""}
+                title={bug.resolution !== "" ? bug.resolution : ""}
+              />))}
+          </ul>}
+
+        {(suggestion.bugs.too_many_open_recent || (suggestion.bugs.too_many_all_others
+            && !suggestion.valid_open_recent)) &&
+            <mark>Exceeded max {BUG_LIMIT} bug suggestions, most of which are likely false positives.</mark>}
+      </li>
+    );
+  }
+}
+
+
+function ListItem(props) {
+  return (
+    <li>
+      <p className="failure-summary-line-empty mb-0">{props.text}</p>
+    </li>
+  );
+}
+
+
+function BugListItem(props) {
+  const {
+    bug, suggestion,
+    bugClassName, title, $timeout, pinboardService, selectedJob,
+  } = props;
+  const bugUrl = getBugUrl(bug.id);
+  const bugSummaryText = escapeHTML(bug.summary);
+  const highlightedTerms = { __html: highlightCommonTerms(bugSummaryText, suggestion.search) };
+
+  return (
+    <li>
+      <button
+        className="btn btn-xs btn-light-bordered"
+        onClick={() => $timeout(() => pinboardService.addBug(bug, selectedJob))}
+        title="add to list of bugs to associate with all pinned jobs"
+      >
+        <i className="fa fa-thumb-tack" />
+      </button>
+      <a
+        className={`${bugClassName} ml-1`}
+        href={bugUrl}
+        target="_blank"
+        rel="noopener"
+        title={title}
+      >{bug.id}
+        <span className={`${bugClassName} ml-1`} dangerouslySetInnerHTML={highlightedTerms} />
```

Comment:
It would be good to get rid of `dangerouslySetInnerHTML` at some point (doesn't need to be now)

Reply:
I agree, yeah.  I have a WIP on this, but it was kind of tricky getting it to work properly.  Using JSX, I could get some matching, but not quite what we currently do.  I'll need to iterate on it a bit.  :)